### PR TITLE
Team1 experiment implementation

### PR DESCRIPTION
**Pull request by sherwin-dc**
Copied same code in #66 into team1 files for experimentation.

Changed agent implementation to support new agent format

Need help with using the new `immutable.Map[commons.ID, HiddenAgentState]` in the agent

### CODE REVIEW

**Comment by sherwin-dc** [comment]
@neeldug I think I'll need your help with `infra/teams/team1/utils/comms.go:33:36` where the interface implementation of Message is not being detected. I am  not sure what I am missing in order to make it work

**Comment by neeldug** [comment]
> @neeldug I think I'll need your help with `infra/teams/team1/utils/comms.go:33:36` where the interface implementation of Message is not being detected. I am not sure what I am missing in order to make it work

Yes, this makes sense since messages must be implemented for all agents and cannot be implemented "per agent", instead it should be done within the `message` package.

**Comment by neeldug** [comment]
> Yes, this makes sense since messages must be implemented for all agents and cannot be implemented "per agent", instead it should be done within the `message` package.

Moreover with #104 hopefully being merged soon, this should make the type system far stricter and ensure specific types of messages are implemented. But essentially implementing message types per-agent shouldn't be possible since the common language needs to be universal

**Comment by sherwin-dc** [comment]
> Moreover with #104 hopefully being merged soon, this should make the type system far stricter and ensure specific types of messages are implemented. But essentially implementing message types per-agent shouldn't be possible since the common language needs to be universal

For our experiment we were planning on having agents send each other messages to share information, so I was looking at creating a message type just for a team experiment.

I can look into doing so, but in the meantime, how would I fill out the `HandleFightRequest` function to create a `message.FightInform` type? I should be able to go from there.
```go
func (r *RandomAgent) HandleFightRequest(_ message.TaggedMessage, _ *immutable.Map[commons.ID, decision.FightAction]) message.FightInform {
	return nil
}
```

**Comment by neeldug** [comment]
> > Moreover with #104 hopefully being merged soon, this should make the type system far stricter and ensure specific types of messages are implemented. But essentially implementing message types per-agent shouldn't be possible since the common language needs to be universal
> 
> For our experiment we were planning on having agents send each other messages to share information, so I was looking at creating a message type just for a team experiment.
> 
> I can look into doing so, but in the meantime, how would I fill out the `HandleFightRequest` function to create a `message.FightInform` type? I should be able to go from there.
> 
> ```go
> func (r *RandomAgent) HandleFightRequest(_ message.TaggedMessage, _ *immutable.Map[commons.ID, decision.FightAction]) message.FightInform {
> 	return nil
> }
> ```

I think you ought to rebase once again, we've changed it a little but now you can see what the server sends as its message and it ought to be identical.

File: pkg/infra/game/message/message.go [hunk 1]
```diff
@@ -48,3 +48,21 @@ func (s StartFight) sealedInform() {
 func (s StartFight) sealedFightInform() {
 	panic("implement me")
 }
+
+type ArrayInfo struct {
```
Comment (neeldug) [inline pkg/infra/game/message/message.go:52]: Could this be moved into a separate file within the same package or even a sub-package just to keep things tidy?

**Review by neeldug — CHANGES_REQUESTED** [review]

File: pkg/infra/game/message/peer2peer.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package message
```
Comment (neeldug) [inline pkg/infra/game/message/peer2peer.go:1]: Could you rename this file to `peer.go`, seems unnecessary to have numbers here.

**Pull request closed without merging**